test/over_react/util/safe_render_manager/safe_render_manager_test.dart

```diff
@@ -0,0 +1,707 @@
+              @required bool verifyImmediateRender,
+              @required bool verifyDeferredRender,
+            }) async {
+              if (verifyImmediateRender && verifyDeferredRender) {
+                throw new ArgumentError('verifyImmediateRender and verifyDeferredRender '
+                    'are mutually exclusive and cannot both be set to true');
+              }
+
+              // Bind guarded so that any failing `expect`s, which result in synchronous errors,
+              // are handled by the test zone's error handler and not swallowed
+              // by whichever zone they end up being run in
+              // (e.g., the root zone in the case of event handlers).
+              final doRenders = bind0Guarded(expectAsync0(() {
+                renderManager.render(Test()(render1Text));
+                // Catch any errors thrown synchronously by this call, which will
+                // happen only when rendering is not deferred...
+                try {
+                  renderManager.render((Test()..onRender = onRender)(render2Text));
+                } on TestExceptionThrownFromLifecycle catch (_) {
+                  // ...but rethrow them if we're not expecting them so they aren't swallowed.
+                  if (!isThrowingTest) rethrow;
+                }
+
+                if (!isThrowingTest) {
+                  if (verifyImmediateRender) {
+                    expect(mountNode.text, render2Text, reason: 'should have updated synchronously');
+                  }
+
+                  if (verifyDeferredRender) {
+                    expect(mountNode.text, isNot(anyOf(render1Text, render2Text)),
+                        reason: 'should have updated synchronously');
+                  }
+                }
+              }, id: 'doRenders', reason: 'needs to be called as part of test'));
+
+              final triggerRenders = setUpAndReturnTriggerRender(doRenders);
+
+              await pumpEventQueue();
+              expect(onMaybeUnmountedCalled, isFalse,
+                  reason: 'test setup: content should still be mounted before doRenders is called');
+              expect(mountNode.text, isNot(anyOf(render1Text, render2Text)),
+                  reason: 'test setup: content should still be mounted before doRenders is called');
+
+              // Catch any errors thrown synchronously by this call, which will
+              // happen only when rendering is deferred...
+              try {
+                triggerRenders();
+              } on TestExceptionThrownFromLifecycle catch (_) {
+                // ...but rethrow them if we're not expecting them so they aren't swallowed.
+                if (!isThrowingTest) rethrow;
+              }
+
+              if (isThrowingTest) {
+                expect(mountNode.text, isEmpty, reason: 'test setup check: React should have unmounted throwing component tree');
```

Comment:
Will we just remove this logic when we port this change into `2.x` stable?

Reply:
I think this is good to keep in as a sanity check